modules/common/webhook/storage.go

```diff
@@ -37,11 +37,27 @@ import (
 //
 //     ValidateStorageRequest(<path>, "500M", "5G", true)
 func ValidateStorageRequest(basePath *field.Path, req string, min string, err bool) (admission.Warnings, field.ErrorList) {
-	storageRequestProd := resource.MustParse(min)
-	storageRequest := resource.MustParse(req)
 	allErrs := field.ErrorList{}
 	allWarn := []string{}
 
+	// If the StorageRequest is a wrong string, we must return
+	// an error. MustParse can't be used in this context as it
+	// generates panic() and we can't recover the operator.
+	storageRequest, parseError := resource.ParseQuantity(req)
+	if parseError != nil {
+		parseQuantityError := fmt.Sprintf("Field %s: %s is invalid",
+			basePath.Child("storageRequest").String(), req)
+		allErrs = append(allErrs, field.Invalid(basePath, req, parseQuantityError))
```

Comment:
@stuggi I'm not sure I'm violating the signature of this function. I always return an error if the input is simply wrong. Considering this function is only used by mariadb-operator, and we might want to add it to glance/swift, I'm wondering if this might be a good time to update it and clarify the expectations of the return (`err` vs `warn`).

Reply:
Ok Thinking more about this I think as per the original design we should honor the warn vs err behavior. I'm going to update it accordingly.